lib/modules/manager/argocd/extract.ts

```diff
@@ -66,28 +101,24 @@ function createDependency(
   };
 }
 
-export function extractPackageFile(
-  content: string,
-  fileName: string,
-  _config?: ExtractConfig
-): PackageFileContent | null {
-  // check for argo reference. API version for the kind attribute is used
-  if (fileTestRegex.test(content) === false) {
-    return null;
+export function processAppSpec(
+  spec: ApplicationSpec | null | undefined
+): PackageDependency[] {
+  if (is.nullOrUndefined(spec)) {
+    return [];
   }
 
-  let definitions: ApplicationDefinition[];
-  try {
-    definitions = loadAll(content) as ApplicationDefinition[];
-  } catch (err) {
-    logger.debug({ err, fileName }, 'Failed to parse ArgoCD definition.');
-    return null;
+  const deps: Array<PackageDependency | null> = [];
+
+  if (is.nonEmptyObject(spec.source)) {
+    deps.push(processSource(spec.source));
   }
 
-  const deps = definitions
-    .filter(is.plainObject)
-    .map((definition) => createDependency(definition))
-    .filter(is.truthy);
+  if (is.array(spec.sources)) {
+    for (const source of coerceArray(spec.sources)) {
+      deps.push(processSource(source));
+    }
+  }
```

Comment:
```suggestion
  for (const source of coerceArray(spec.sources)) {
    deps.push(processSource(source));
  }
```
The if clause is not needed in case `coerceArray` is used.